src/compiler/transformers/es2015.ts

```diff
@@ -2569,62 +2566,54 @@ namespace ts {
          * @param node An ObjectLiteralExpression node.
          */
         function visitObjectLiteralExpression(node: ObjectLiteralExpression): Expression {
-            // We are here because a ComputedPropertyName was used somewhere in the expression.
             const properties = node.properties;
-            const numProperties = properties.length;
 
             // Find the first computed property.
             // Everything until that point can be emitted as part of the initial object literal.
-            let numInitialProperties = numProperties;
-            let numInitialPropertiesWithoutYield = numProperties;
-            for (let i = 0; i < numProperties; i++) {
+            let numInitialProperties = -1, hasComputed = false;
+            for (let i = 0; i < properties.length; i++) {
                 const property = properties[i];
-                if ((property.transformFlags & TransformFlags.ContainsYield && hierarchyFacts & HierarchyFacts.AsyncFunctionBody)
-                    && i < numInitialPropertiesWithoutYield) {
-                    numInitialPropertiesWithoutYield = i;
-                }
-                if (Debug.checkDefined(property.name).kind === SyntaxKind.ComputedPropertyName) {
+                hasComputed = Debug.checkDefined(property.name).kind === SyntaxKind.ComputedPropertyName;

```

Comment:
Note that https://github.com/microsoft/TypeScript/pull/38880/files#diff-e060748c3972d7c43ccd394a12566242R186 still has the excess indentation. I would try this instead:
```ts
if (property.transformFlags & TransformFlags.ContainsYield && hierarchyFacts & HierarchyFacts.AsyncFunctionBody) {
  numInitialProperties = i;
  break;
}
if (Debug.checkDefined(property.name).kind === SyntaxKind.ComputedPropertyName) {
  numInitialProperties = i;
  hasComputed = true;
  break;
}
```
This we we only add the indentation if we aren't yielding.